### PR TITLE
Bump to 0.3.1, bump stack resolver, add Nix flake

### PR DESCRIPTION
**Pull request by centromere**

### CODE REVIEW

**Comment by chris-martin** [comment]
Hmm, now yet another build system doesn't like the license?

```
cabal2nix: Distribution.Nixpkgs.Haskell.FromCabal.License.fromCabalLicense: unknown licenseUnknownLicense "Unlicense"
Choose one of: GPL, GPL-2, GPL-3, LGPL, LGPL-2.1, LGPL-3, AGPL, AGPL-3, BSD2, BSD3, MIT, ISC, MPL-2.0, Apache, Apache-2.0, PublicDomain, AllRightsReserved, OtherLicense
CallStack (from HasCallStack):
  error, called at src/Distribution/Nixpkgs/Haskell/FromCabal/License.hs:46:62 in cabal2nix-2.19.1-8s0wgK1XwOXL0JsDOh92VU:Distribution.Nixpkgs.Haskell.FromCabal.License
```

I don't really understand what's going on here, because...

nixpkgs is aware of Unlicense https://github.com/NixOS/nixpkgs/blob/nixos-22.11/lib/licenses.nix#L916-L919

and so is cabal2nix https://github.com/NixOS/cabal2nix/blob/0d1a5a7d227683e9d099665f645a8618ec48ef94/cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/License.hs#L144

**Comment by chris-martin** [comment]
Nevermind, figured it out - I neglected to raise the `cabal-version` field when I updated the license. Fixed by  #13

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  description = "doorctl";
```
Comment (chris-martin) [inline flake.nix]: What's doorctl, is this copied over from another project?

**Comment by chris-martin** [comment]
Sorry, although I do use Nix a lot, I really don't know much about Nix flakes - What's the objective here? Is it for testing, or does the flake provide some additional utility for nix users?

File: flake.nix (hunk 1 above)
Comment (centromere) [reply flake.nix]: Oops, yes.

**Comment by centromere** [comment]
> Sorry, although I do use Nix a lot, I really don't know much about Nix flakes - What's the objective here? Is it for testing, or does the flake provide some additional utility for nix users?

The objective is to make it easier for Nix users to pull the package in and work on it.

**Comment by centromere** [comment]
@chris-martin Are you okay with this now?

**Comment by chris-martin** [comment]
Much belated :+1:, I'm down with the flakes now and it's great.

**Pull request merged**